app/javascript/react/store/sensorsSlice.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  PayloadAction,
+  createAsyncThunk,
+  createSelector,
+  createSlice,
+} from "@reduxjs/toolkit";
+import { AxiosResponse } from "axios";
+import { RootState } from ".";
+import { oldApiClient } from "../api/apiClient";
+import { API_ENDPOINTS } from "../api/apiEndpoints";
+import { StatusEnum } from "../types/api";
+import { SessionType } from "../types/filters";
+import { Sensor } from "../types/sensors";
+import { getErrorMessage } from "../utils/getErrorMessage";
+
+interface SensorsState {
+  sensors: Sensor[];
+  fetchSensorsStatus: StatusEnum;
+}
+
+const initialState: SensorsState = {
+  sensors: [],
+  fetchSensorsStatus: StatusEnum.Idle,
+};
+
+export const fetchSensors = createAsyncThunk(
+  "sensors",
+  async (sessionType: SessionType, { rejectWithValue }) => {
+    const sessionTypeCapitalized =
+      sessionType[0].toUpperCase() + sessionType.slice(1);
```

Comment:
Why do you need to do this? What does `.slice(1)` do? 😅

Reply:
I think this is to ensure that when the sessionType is passed to enpoint it’s in the correct format.